src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/EServiceApiServiceImpl.scala

```diff
@@ -413,10 +415,24 @@ class EServiceApiServiceImpl(
       descriptor: CatalogDescriptor,
       updateEServiceDescriptorState: CatalogDescriptorState
     ): CatalogDescriptor = {
-      val activatedAt: Option[OffsetDateTime] =
-        if (descriptor.state == Draft && updateEServiceDescriptorState == Published) offsetDateTimeSupplier.get().some
-        else descriptor.activatedAt
-      descriptor.copy(state = updateEServiceDescriptorState, activatedAt = activatedAt)
+      (descriptor.state, updateEServiceDescriptorState) match {
+        case (Draft, Published)      =>
+          descriptor.copy(state = updateEServiceDescriptorState, publishedAt = offsetDateTimeSupplier.get().some)
+        case (Published, Suspended)  =>
+          descriptor.copy(state = updateEServiceDescriptorState, suspendedAt = offsetDateTimeSupplier.get().some)
+        case (Suspended, Published)  =>
+          descriptor.copy(state = updateEServiceDescriptorState, suspendedAt = None)
+        case (Suspended, Deprecated) =>
+          descriptor.copy(
+            state = updateEServiceDescriptorState,
+            suspendedAt = None,
+            deprecatedAt = offsetDateTimeSupplier.get().some
+          )
+        // case (Published, Archived) => Archived logic not yet implemented
+        case (Published, Deprecated) =>
+          descriptor.copy(state = updateEServiceDescriptorState, deprecatedAt = offsetDateTimeSupplier.get().some)
+        case _                       => descriptor.copy(state = updateEServiceDescriptorState) // Solve this case in PR
```

Comment:
I asked to @bloodsky to put this comment.
What should be the proper behavior in the case of _?
Probably we should raise an error here. What do you think @galales and @TonioGela ?

Reply:
I think it's safer to blindly copy as it is now.
This pattern matching is focused on settings dates, if there are cases not needed here my opinion is that we should passthrough.